mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -420,41 +449,77 @@ def _get_datetimes(ds, timeVariableName, yearOffset):  # {{{
     02/16/2017
     """
 
-    if isinstance(timeVariableName, (tuple, list)):
+    if isinstance(inTimeVariableName, (tuple, list)):
         # we want to average the two
-        assert(len(timeVariableName) == 2)
-        starts = _get_datetimes(ds, timeVariableName[0], yearOffset)
-        ends = _get_datetimes(ds, timeVariableName[1], yearOffset)
-        datetimes = [starts[i] + (ends[i] - starts[i])/2
-                     for i in range(len(starts))]
-        return datetimes
-
-    timeVariable = ds[timeVariableName]
-
-    if timeVariable.dtype == '|S64':
-        # this is a variable like date strings like 'xtime'
-        time = [''.join(atime).strip() for atime in timeVariable.values]
-        datetimes = [datetime.datetime(yearOffset + int(x[:4]),
-                                       int(x[5:7]), int(x[8:10]),
-                                       int(x[11:13]), int(x[14:16]),
-                                       int(x[17:19]))
-                     for x in time]
-    elif timeVariable.dtype == 'float64':
-        # this array contains floating-point days like
-        # 'daysSinceStartOfSim'
-        start = datetime.datetime(year=yearOffset+1, month=1, day=1)
-        datetimes = [start + datetime.timedelta(x)
-                     for x in timeVariable.values]
-    elif timeVariable.dtype == 'timedelta64[ns]':
-        # this array contains a variable like 'daysSinceStartOfSim' as a
-        # timedelta64
-        start = datetime.datetime(year=yearOffset+1, month=1, day=1)
-        datetimes = [start + x for x in
-                     pd.to_timedelta(timeVariable.values, unit='ns')]
+        assert(len(inTimeVariableName) == 2)
+
+        dsStart = _parse_dataset_time(
+            ds=ds,
+            inTimeVariableName=inTimeVariableName[0],
+            calendar=calendar,
+            simulationStartTime=simulationStartTime,
+            outTimeVariableName=outTimeVariableName,
+            referenceDate=referenceDate)
+        dsEnd = _parse_dataset_time(
+            ds=ds,
+            inTimeVariableName=inTimeVariableName[1],
+            calendar=calendar,
+            simulationStartTime=simulationStartTime,
+            outTimeVariableName=outTimeVariableName,
+            referenceDate=referenceDate)
+        starts = dsStart[outTimeVariableName].values
+        ends = dsEnd[outTimeVariableName].values
+
+        # replace the time in starts with the mean of starts and ends
+        dsStart[outTimeVariableName] = [starts[i] +
+                                        (ends[i] - starts[i])/2
+                                        for i in range(len(starts))]
+
+        return dsStart
+
```

Comment:
Do you mean by "this" the remainder of the function or the return statement?  The first part of the function is for cases when there are two time variables to be averaged.  The remainder of the function is for cases when there is just one time variable (either because we're recursively calling the function or because we're not averaging).  I don't think the fact that the types are '|S64' and 'float64' is particularly the relevant piece.